session/session.go

```diff
@@ -1159,10 +1159,66 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 	return prepareExec.ID, prepareExec.ParamCount, prepareExec.Fields, nil
 }
 
+// CachedPlanExec short path ONLY for cached "point select plan" execution
+func (s *session) CachedPlanExec(ctx context.Context,
+	stmtID uint32, prepared *ast.Prepared, args []types.Datum) (sqlexec.RecordSet, error) {
+	is := executor.GetInfoSchema(s)
+	execAst := &ast.ExecuteStmt{ExecID: stmtID}
+	if err := executor.ResetContextOfStmt(s, execAst); err != nil {
+		return nil, err
+	}
+	execAst.BinaryArgs = args
+	execPlan, err := plannercore.OptimizeExecStmt(ctx, s, execAst, is)
+	if err != nil {
+		return nil, err
+	}
+	stmt := &executor.ExecStmt{
+		InfoSchema:  is,
+		Plan:        execPlan,
+		StmtNode:    prepared.Stmt,
+		Ctx:         s,
+		OutputNames: execPlan.OutputNames(),
+	}
+	rs, err := stmt.GetPointRecord(ctx, is, s)
+	s.txn.changeToInvalid()
+	return rs, err
+}
+
 // ExecutePreparedStmt executes a prepared statement.
 func (s *session) ExecutePreparedStmt(ctx context.Context, stmtID uint32, args []types.Datum) (sqlexec.RecordSet, error) {
-	s.PrepareTxnCtx(ctx)
+	var err error
 	s.sessionVars.StartTime = time.Now()
+	prepared, ok := s.sessionVars.PreparedStmts[stmtID]
+	if !ok {
+		err = errors.Errorf("Prepared statement not found")
+		logutil.Logger(ctx).Error("prepared not found", zap.Uint32("stmtID", stmtID))
+		return nil, err
+	}
+	if prepared.CachedPlan != nil {
+		cachedValue := plannercore.PSTMTPlanCacheValue{}
```

Comment:
this `cachedValue` created but never be use?

Reply:
not used currently